src/libs/actions/Report.js

```diff
@@ -1543,6 +1626,27 @@ function renameReport(reportID, reportName) {
         .finally(() => Onyx.set(ONYXKEYS.IS_LOADING_RENAME_POLICY_ROOM, false));
 }
 
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+    callback: (reportAction) => {
+        // // If chat report receives an action with IOU and we have an IOUReportID, update IOU object
+        // if (reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && reportAction.originalMessage.IOUReportID) {
+        //     const iouReportID = reportAction.originalMessage.IOUReportID;
+
+        //     // If the IOUDetails object exists we are in the Send Money flow, and we should not fetch and update the chatReport
+        //     // as this would overwrite any existing IOUs. For all other cases we must update the chatReport with the iouReportID as
+        //     // if we don't, new IOUs would not be displayed and paid IOUs would still show as unpaid.
+        //     if (reportAction.originalMessage.IOUDetails === undefined) {
+        //         fetchIOUReportByIDAndUpdateChatReport(iouReportID, reportID);
+        //     }
+        // }
```

Comment:
I know this is commented out, but I think this train of thought is wrong.

If we are sending out IOU actions then we could also send the IOU report data with that action. I'm not sure why we have logic that:

1. Triggers one kind of data to be sent
2. The client then requests more related data, but only sometimes depending on the flow

Ideally the client doesn't have to do anything and we just push the right data to it...

cc @Julesssss @mountiny

Reply:
cc @danieldoglas I think you were asking about something similar [here](https://github.com/Expensify/Expensify/issues/212809#issuecomment-1145366512). We should be already sending all this data as part of the `Report_GetHistory`.

Reply:
This could be for a new report action that we haven't fetched the history for yet. It might need to happen in `Report_GetHistory`, but would have to happen when a new report was added as well.

Example:

- User has no report or report actions for a chat
- A new IOU action appears
- This code will fetch the IOU report

So, I think we need to send the IOU report with the action if it exists.